src/power_grid_model_io/utils/excel_ambiguity_checker.py

```diff
@@ -0,0 +1,155 @@
+            shared_strings (list): A list of shared strings extracted from the Excel file.
+
+        Returns:
+            list: A list of column names found in the row.
+        """
+        column_names = []
+        for c in row.findall(".//c", namespaces=XML_NAME_SPACE):
+            cell_type = c.get("t")
+            value = c.find(".//v", namespaces=XML_NAME_SPACE)
+            if cell_type == "s" and value is not None:
+                column_names.append(shared_strings[int(value.text)])
+            elif value is not None:
+                column_names.append(value.text)
+            else:
+                column_names.append(None)
+        return column_names
+
+    def _parse_excel_file(self) -> None:
+        """
+        Parses the Excel file to extract sheet names and their corresponding column names.
+        """
+        with zipfile.ZipFile(self._file_path) as z:
+            shared_strings = self._parse_zip(z)
+            workbook_xml = z.read("xl/workbook.xml")
+            xml_tree = ET.fromstring(workbook_xml)
+            sheets = xml_tree.findall(".//sheet", namespaces=XML_NAME_SPACE)
+
+            for index, sheet in enumerate(sheets, start=1):
+                sheet_name = str(sheet.get("name"))
+                sheet_file_path = f"xl/worksheets/sheet{index}.xml"
+
+                with z.open(sheet_file_path) as f:
+                    sheet_tree = ET.parse(f)
+                    rows = sheet_tree.findall(".//row", namespaces=XML_NAME_SPACE)
+                    if rows:
+                        column_names = self._get_column_names_from_row(rows[self._col_name_in_row], shared_strings)
+                        self.sheets[sheet_name] = [name for name in column_names if name is not None]
+
+    def list_sheets(self) -> List[str]:
+        """
+        Get the list of all sheet names in the Excel file.
+
+        Returns:
+            List[str]: list of all sheet names
+        """
+        return list(self.sheets.keys())
+
+    def check_ambiguity(self) -> Tuple[bool, Dict[str, List[str]]]:
+        """
+        Check if there is ambiguity in column names across sheets.
+
+        Returns:
+            Tuple[bool, Dict[str, List[str]]]: A tuple containing a boolean indicating if any ambiguity was found,
+            and a dictionary with sheet names as keys and lists of ambiguous column names as values.
+        """
+        res: Dict[str, List[str]] = {}
+        if not self._valid_file:
+            return False, res
+        for sheet_name, column_names in self.sheets.items():
+            column_name_counts = Counter(column_names)
+            duplicates = [name for name, count in column_name_counts.items() if count > 1]
+            if duplicates:
+                res[sheet_name] = duplicates
+        return bool(res), res
+
+
+# Example usage
+if __name__ == "__main__":
+    excel_file_checker = ExcelAmbiguityChecker("data.xlsx")
```

Comment:
do we have more of this stuff in  PGM or is this a first?

Reply:
I think first since we depend on pandas and openpyxl

Reply:
I will comment out this section btw when cleaning